### PR TITLE
Setup Github Action CICD

### PR DESCRIPTION
**Pull request by hochiw**
## Features
- Automatically run ci tests on pull request open/update
- Build and push images on commit push to master/staging branch
- Notify deployment repository for deployment on image push

Tested by opening a branch on the oursky repo, existing deployment is uninstalled manually and redeployed by the deployment repo to ensure it's working correctly.

based on #15 #16 #17 
refs #12

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -9,9 +9,12 @@ env:
...
   DEPLOYMENT_WEBHOOK: ${{ secrets.DEPLOYMENT_WEBHOOK }}
 jobs:
-  build-and-push-images:
-    name: Build And Push Docker Images
+  build-and-push-image:
+    name: Build And Push ${{ matrix.target }} Image
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        target: [react-app, graphql-server, bdjuno]
```
Comment (chihimng) [inline .github/workflows/build.yml:17]: 👍

**Review by chihimng — APPROVED** [review]
we might need an example of the deployment repo / guides to setup ci

**Comment by chihimng** [comment]
plz rebase 🙏

**Comment by hochiw** [comment]
@chihimng rebased

**Pull request merged**